### DISCUSSION TITLE
How to render and type `fields.mdx.inline`

### DISCUSSION THREAD

**Discussion by tordans** [Thinkmill/keystatic#1318] [Q&A]
I am using `content: fields.mdx.inline({ label: 'Content' }),` with `format: { data: 'json' },` so keystatic puts all my data in a `.json` file.

I am having two issues with this:

- The `content` property does not show up in the astro types.
- I cannot use `const { Content } = await page.render()` anymore and I don't know how to render the raw mdx I receive.

How does this work?

Docs: https://keystatic.com/docs/fields/mdx#inline

**Comment by tordans** [comment]
My current workaround is, to [use **micromark**](https://github.com/micromark/micromark/tree/main?tab=readme-ov-file#use) which is already part of [keystatic](https://github.com/search?q=repo%3AThinkmill%2Fkeystatic%20micromark&type=code) to render the inline Markdown manually, which works fine.

However, tried this only for mdx where images and custom components are disabled. I assume adding those would cause issues.

**Reply by tordans** [reply]
Examples:
- https://github.com/FixMyBerlin/nudafa/blob/main/src/components/page_massnahmen/dashboard/components/Markdown.tsx
- https://github.com/search?q=repo%3AFixMyBerlin%2Fradinfra.de%20micromark&type=code (without adding it as package explicitly)

**Comment by amirzezo201** [comment]
@tordans  hey torden iam also using markdoc.inline which works really good in local but on netlify deployment it doenst even open the collection entry which have markdoc.inline block
